extensions/resource-deployment/src/services/resourceTypeService.ts

```diff
@@ -237,14 +271,24 @@ export class ResourceTypeService implements IResourceTypeService {
 	private getOkButtonText(resourceType: ResourceType, selectedOptions: { option: string, value: string }[]): string | undefined {
 		if (resourceType.okButtonText) {
 			for (const possibleOption of resourceType.okButtonText) {
-				if (processWhenClause(possibleOption.when, selectedOptions)) {
+				if (possibleOption && processWhenClause(possibleOption.when, selectedOptions)) {
 					return possibleOption.value;
 				}
 			}
 		}
 		return loc.select;
 	}
 
+	private getAgreementInfo(resourceType: ResourceType, selectedOptions: { option: string, value: string }[]): AgreementInfo | undefined {
+		if (resourceType.agreement) {
+			for (const possibleOption of resourceType.agreement) {
+				if (possibleOption && processWhenClause(possibleOption.when, selectedOptions)) {
```

Comment:
Same here - when would possibleOption be undefined (when we know we have an agreement array already)